### PR TITLE
Adding PostAction plugin hook

### PR DESCRIPTION
**Pull request by jespino**
#### Summary
Add a new extension point in the plugins to allow to register post actoins, that are new items in the post that can be a button or a menu

#### Ticket Link
mattermost/mattermost-plugin-ai#35

#### Screenshots
imagen
imagen

#### Release Note

```release-note
Added Post Action plugin hook to allow plugins to register new items in the post menu
Added Post Editor Action plugin hook to allow plugins to register new items in the post editor menu
```

### CODE REVIEW

File: webapp/channels/src/plugins/registry.ts [hunk 1]
```diff
@@ -126,6 +126,18 @@ export default class PluginRegistry {
         return dispatchPluginComponentAction('PostMessageAttachment', this.id, component);
     });
 
+    // Register a component to the add to the post message menu.
+    // Accepts a React component. Returns a unique identifier.
+    registerPostActionComponent = reArg(['component'], ({component}: DPluginComponentProp) => {
```
Comment (lieut-data) [inline webapp/channels/src/plugins/registry.ts:131]: If we're not going to add this to the bottom, can we locate this "closer" to the existing `registerPostDropdownMenuAction` and clarify the difference between the two types of registrations?

File: webapp/channels/src/plugins/registry.ts [hunk 2]
```diff
@@ -126,6 +126,18 @@ export default class PluginRegistry {
...
+        return dispatchPluginComponentAction('PostAction', this.id, component);
+    });
+
+    // Register a component to the add to the post editor menu.
+    // Accepts a React component. Returns a unique identifier.
+    registerPostEditorActionComponent = reArg(['component'], ({component}: DPluginComponentProp) => {
```
Comment (lieut-data) [inline webapp/channels/src/plugins/registry.ts:137]: nit: `PostEditor` makes it sound like this appears where I can edit posts, but I think this only works for `PostCreator` at the present?

**Review by lieut-data — CHANGES_REQUESTED** [review]
Thanks, @jespino! Just two small comments below, but I'm also wondering how we can automate the testing of this to reduce QA overhead. Is there a planned update to the demo plugin?

File: webapp/channels/src/plugins/registry.ts (hunk 2 above)
Comment (jespino) [reply webapp/channels/src/plugins/registry.ts:137]: It is adding the functionallity to the post advanced editor, that I would expect, eventually to be use for editing posts too. I think this name is correct.

File: webapp/channels/src/plugins/registry.ts (hunk 1 above)
Comment (jespino) [reply webapp/channels/src/plugins/registry.ts]: I moved that right before the DropdownMenuAction registry function and slightly changed the comments. PTAL.

**Review by lieut-data — APPROVED** [review]

**Comment by lindy65** [comment]
Hey @jespino - I've spun up a test server but uploading plugins to spinwicks is disabled and needs to be set in config.json.  Can you upload the plugin for me please?  
image

**Comment by jespino** [comment]
@lindy65 I'm not able to upload the plugin, lets talk tomorrow and see how we can get this tested.

**Comment by lindy65** [comment]
I'm out tomorrow @jespino but back on Monday 👍

**Review by lindy65 — APPROVED** [review]
Thanks @jespino - tested on the server you spun up and didn't find any issues.  Created a [Jira ticket](https://mattermost.atlassian.net/browse/MM-53954) to track / test after merging

**Comment by cwarnermm** [comment]
@jespino - Are there any Developer Documentation updates needed for this PR?

**Pull request merged**